### PR TITLE
refactor: separate integration test package (Step 1)

### PR DESCRIPTION
**Pull request by keyvankhademi**
This PR moves the integration tests into a separate cargo package. It's almost copied as is with some minor refactors.

Refactor 1: app the pda address calculations moved to a pda.rs file in the respective program folder
Refactor 2: some instructions moved to a instructions.rs file in the respective program folder

The idea is to eventually move all the instructions in instructions.rs but separated via parent folders with respect to the related program.

### CODE REVIEW

File: staking/programs/integrity-pool/Cargo.toml [hunk 1]
```diff
@@ -21,12 +21,4 @@ publisher-caps = {path = "../publisher-caps", features = ["no-entrypoint"]}
 
 
 [dev-dependencies]
-litesvm = "0.1.0"
-solana-program = { workspace = true }
 solana-sdk = { workspace = true }
```
Comment (guibescos) [inline staking/programs/integrity-pool/Cargo.toml:24]: do we still need this?

File: staking/integration-tests/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+pub mod integrity_pool;
+pub mod publisher_caps;
+pub mod setup;
+pub mod staking;
+pub mod utils;
```
Comment (guibescos) [inline staking/integration-tests/src/lib.rs:5]: what's the benefit of making this a lib.rs vs a mod.rs

File: staking/programs/integrity-pool/Cargo.toml (hunk 1 above)
Comment (keyvankhademi) [reply staking/programs/integrity-pool/Cargo.toml:24]: its used in the tests
```
 assert!(
            solana_sdk::borsh0_10::get_packed_len::<SlashEvent>()
                + SlashEvent::discriminator().len()
                == SlashEvent::LEN
        );
```

**Review by guibescos — APPROVED** [review]
lgtm, thanks

File: staking/programs/integrity-pool/Cargo.toml (hunk 1 above)
Comment (keyvankhademi) [reply staking/programs/integrity-pool/Cargo.toml:24]: I can replace it with `anchor_lang::solana_program::borsh0_10`

Comment (keyvankhademi) [reply staking/programs/integrity-pool/Cargo.toml]: fixed

**Pull request merged**